From 708ba67b9bf141c4e4776b1a9a7c4d7cf400de47 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Fri, 16 May 2025 11:26:31 +0200
Subject: [PATCH] server: Fix shared object offset when additional blocks are
 allocated.

The shared object offset is an absolute offset from the start of the
session shared memory file, as it is used to identify the corresponding
mmapped block on the client side.
---
 server/mapping.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/mapping.c b/server/mapping.c
index 83538831ea2..f681704bec9 100644
--- a/server/mapping.c
+++ b/server/mapping.c
@@ -1394,7 +1394,7 @@ volatile void *alloc_shared_object(void)
 
         if (!(block = find_free_session_block( size ))) return NULL;
         object = (struct session_object *)(block->data + block->used_size);
-        object->offset = (char *)&object->obj - block->data;
+        object->offset = block->offset + (char *)&object->obj - block->data;
         block->used_size += size;
     }
 
-- 
GitLab

